crates/rooch-types/src/transaction/authenticator.rs

```diff
@@ -382,342 +209,10 @@ impl fmt::Display for Authenticator {
-    /// Create a random authentication key. For testing only
-    pub fn random() -> Self {
-        let mut rng = OsRng;
-        let buf: [u8; Self::LENGTH] = rng.gen();
-        AuthenticationKey::new(buf)
-    }
-
-    /// Check the auth key is dummy empty key
-    pub fn is_dummy(&self) -> bool {
-        self.0 == Self::DUMMY_KEY
-    }
-}
-
-impl ValidCryptoMaterial for AuthenticationKey {
-    fn to_bytes(&self) -> Vec<u8> {
-        self.to_vec()
-    }
-}
-
-/// A value that can be hashed to produce an authentication key
-pub struct AuthenticationKeyPreimage(Vec<u8>);
-
-impl AuthenticationKeyPreimage {
-    /// Return bytes for (public_key | scheme_id)
-    fn new(mut public_key_bytes: Vec<u8>, scheme: BuiltinScheme) -> Self {
-        public_key_bytes.push(scheme as u8);
-        Self(public_key_bytes)
-    }
-
-    /// Construct a preimage from an Ed25519 public key
-    pub fn ed25519(public_key: &Ed25519PublicKey) -> AuthenticationKeyPreimage {
-        Self::new(public_key.to_bytes().to_vec(), BuiltinScheme::Ed25519)
-    }
-
-    /// Construct a preimage from a MultiEd25519 public key
-    pub fn multi_ed25519(public_key: &MultiEd25519PublicKey) -> AuthenticationKeyPreimage {
-        Self::new(public_key.to_bytes(), BuiltinScheme::MultiEd25519)
-    }
-
-    /// Construct a vector from this authentication key
-    pub fn into_vec(self) -> Vec<u8> {
-        self.0
-    }
-}
-
-impl TryFrom<&[u8]> for AuthenticationKey {
-    type Error = CryptoMaterialError;
-
-    fn try_from(bytes: &[u8]) -> std::result::Result<AuthenticationKey, CryptoMaterialError> {
-        if bytes.len() != Self::LENGTH {
-            return Err(CryptoMaterialError::WrongLengthError);
-        }
-        let mut addr = [0u8; Self::LENGTH];
-        addr.copy_from_slice(bytes);
-        Ok(AuthenticationKey(addr))
-    }
-}
-
-impl TryFrom<Vec<u8>> for AuthenticationKey {
-    type Error = CryptoMaterialError;
-
-    fn try_from(bytes: Vec<u8>) -> std::result::Result<AuthenticationKey, CryptoMaterialError> {
-        AuthenticationKey::try_from(&bytes[..])
-    }
-}
-
-impl FromStr for AuthenticationKey {
-    type Err = Error;
-
-    fn from_str(s: &str) -> Result<Self> {
-        ensure!(
-            !s.is_empty(),
-            "authentication key string should not be empty.",
-        );
-        Ok(AuthenticationKey::from_encoded_string(s)?)
-    }
-}
-
-impl AsRef<[u8]> for AuthenticationKey {
-    fn as_ref(&self) -> &[u8] {
-        &self.0
-    }
-}
-
-impl fmt::LowerHex for AuthenticationKey {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", hex::encode(self.0))
-    }
-}
-
-impl fmt::Display for AuthenticationKey {
-    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
-        // Forward to the LowerHex impl with a "0x" prepended (the # flag).
-        write!(f, "0x{:#x}", self)
-    }
-}
-
-#[derive(Clone, Debug, Hash, PartialEq, Eq, DeserializeKey, SerializeKey)]
-pub enum AccountPublicKey {
```

Comment:
The `AccountPublicKey` and `AccountPrivateKey` are suitable wrappers for supporting multi schemes. We can consider keeping them or introducing them again in the future PR.

Reply:
Later, when implementing multi-schemes, I will consider bringing it back.